### PR TITLE
Add color support to Orbit API macros in Orbit.h

### PR DESCRIPTION
**Pull request by pierricgimmig**
- User can specify the color of timeslices and tracked variables
- Add a set of predefined colors based on Google's Material Design
- Original macros still work

image

### CODE REVIEW

File: OrbitGl/ThreadTrack.cpp [hunk 1]
```diff
@@ -80,6 +80,30 @@ bool ThreadTrack::IsTimerActive(const TimerInfo& timer_info) const {
   return GOrbitApp->IsFunctionVisible(timer_info.function_address());
 }
 
+static inline Color ToColor(uint64_t val) {
```
Comment (reichlfl) [inline OrbitGl/ThreadTrack.cpp:83]: [[nodiscard]]

File: OrbitGl/ThreadTrack.cpp [hunk 2]
```diff
@@ -80,6 +80,30 @@ bool ThreadTrack::IsTimerActive(const TimerInfo& timer_info) const {
...
+  return Color((val >> 24) & 0xFF, (val >> 16) & 0xFF, (val >> 8) & 0xFF, val & 0xFF);
+}
+
+static inline std::optional<Color> GetUserColor(const TimerInfo& timer_info,
```
Comment (reichlfl) [inline OrbitGl/ThreadTrack.cpp:87]: [[nodiscard]]

**Review by reichlfl — APPROVED** [review]
Very cool! Minor comments, otherwise LGTM. I don't know (yet) how the API part works, so I can't judge the effect of this, you should probably get another lgtm from someone who knows better.

File: OrbitGl/ThreadTrack.cpp (hunk 1 above)
Comment (pierricgimmig) [reply OrbitGl/ThreadTrack.cpp]: Done.

File: OrbitGl/ThreadTrack.cpp (hunk 2 above)
Comment (pierricgimmig) [reply OrbitGl/ThreadTrack.cpp]: Done.

File: OrbitGl/ThreadTrack.cpp [hunk 3]
```diff
@@ -80,6 +80,30 @@ bool ThreadTrack::IsTimerActive(const TimerInfo& timer_info) const {
   return GOrbitApp->IsFunctionVisible(timer_info.function_address());
 }
 
+[[nodiscard]] static inline Color ToColor(uint64_t val) {
+  return Color((val >> 24) & 0xFF, (val >> 16) & 0xFF, (val >> 8) & 0xFF, val & 0xFF);
+}
+
+[[nodiscard]] static inline std::optional<Color> GetUserColor(const TimerInfo& timer_info,
+                                                              const FunctionInfo* function_info) {
+  if (function_info != nullptr) {
```
Comment (dimitry-) [inline OrbitGl/ThreadTrack.cpp:89]: if (function_info == nullptr) {
  return {}; // Also this probably should be a check.
}

File: OrbitGl/ThreadTrack.cpp [hunk 4]
```diff
@@ -80,6 +80,30 @@ bool ThreadTrack::IsTimerActive(const TimerInfo& timer_info) const {
...
+    FunctionInfo::OrbitType type = function_info->type();
+    if (type == FunctionInfo::kOrbitTimerStart || type == FunctionInfo::kOrbitTimerStartAsync) {
```
Comment (dimitry-) [inline OrbitGl/ThreadTrack.cpp:91]: again - reverse the if here.

File: OrbitGl/ThreadTrack.cpp [hunk 5]
```diff
@@ -80,6 +80,30 @@ bool ThreadTrack::IsTimerActive(const TimerInfo& timer_info) const {
   return GOrbitApp->IsFunctionVisible(timer_info.function_address());
 }
 
+[[nodiscard]] static inline Color ToColor(uint64_t val) {
+  return Color((val >> 24) & 0xFF, (val >> 16) & 0xFF, (val >> 8) & 0xFF, val & 0xFF);
+}
+
+[[nodiscard]] static inline std::optional<Color> GetUserColor(const TimerInfo& timer_info,
+                                                              const FunctionInfo* function_info) {
```
Comment (dimitry-) [inline OrbitGl/ThreadTrack.cpp:88]: I am not sure if inline gives you anything here.

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: OrbitGl/ThreadTrack.cpp (hunk 3 above)
Comment (pierricgimmig) [reply OrbitGl/ThreadTrack.cpp:89]: Done.

File: OrbitGl/ThreadTrack.cpp (hunk 4 above)
Comment (pierricgimmig) [reply OrbitGl/ThreadTrack.cpp:91]: Done.

**Pull request merged**